src/test/integration/fastapi/tests/test_fastapi.py

```diff
@@ -48,32 +47,74 @@ def test_200_OK(self):
                 )
             )
 
+    def test_endpoint_filter_match(self):
+        endpoint = f"http://localhost:{APP_PORT}/"
+        with FastApiApp(
+            "app:app",
+            APP_PORT,
+            {"LUMIGO_AUTO_FILTER_HTTP_ENDPOINTS_REGEX": ".*(localhost|127.0.0.1).*/$"},
+        ):
+            response = requests.get(endpoint)
+            response.raise_for_status()
+            body = response.json()
+            self.assertEqual(body, {"message": "Hello FastAPI!"})
+
+            response = requests.get(endpoint + "unmatched")
+            response.raise_for_status()
+            body = response.json()
+            self.assertEqual(body, {"message": "Hello again, FastAPI!"})
+
+            spans_container = SpansContainer.get_spans_from_file(
+                wait_time_sec=10, expected_span_count=3
+            )
+            self.assertEqual(3, len(spans_container.spans))
+
+    def test_endpoint_filter_no_match(self):
+        endpoint = f"http://localhost:{APP_PORT}/"
+        with FastApiApp(
+            "app:app",
+            APP_PORT,
+            {"LUMIGO_AUTO_FILTER_HTTP_ENDPOINTS_REGEX": ".*not_matching.*"},
+        ):
+            response = requests.get(endpoint)
+            response.raise_for_status()
+            body = response.json()
+            self.assertEqual(body, {"message": "Hello FastAPI!"})
+
+            response = requests.get(endpoint + "unmatched")
+            response.raise_for_status()
+            body = response.json()
+            self.assertEqual(body, {"message": "Hello again, FastAPI!"})
+
+            spans_container = SpansContainer.get_spans_from_file(
+                wait_time_sec=10, expected_span_count=6
+            )
+            self.assertEqual(6, len(spans_container.spans))
+
     def test_requests_instrumentation(self):
         with FastApiApp("app:app", APP_PORT):
             response = requests.post(
                 f"http://localhost:{APP_PORT}/invoke-requests", json={"a": "b"}
             )
             response.raise_for_status()
-
             body = response.json()
-
             expected_url = "http://sheker.kol.shehoo:8021/little-response"
-
             self.assertIn(expected_url, body["url"])
 
             spans_container = SpansContainer.get_spans_from_file(
-                wait_time_sec=10, expected_span_count=4
+                wait_time_sec=10, expected_span_count=5
```

Comment:
who's the extra one?

Reply:
i don't remember - i believe this is from when we updated the dependencies last month. the mismatch between the count in this line and the line that follows has no effect locally, but introduces flakiness into the CI testing